app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -218,182 +228,294 @@ private fun MessageComposer(
+        // because it is "floating" freely with an absolute x-y position inside of the ConstrainLayout
+        // wrapping the whole content when in the FullScreen state we are giving it max height
+        // when in active state we limit the height to max 82.dp
+        // other we let it wrap the content of the height, which will be equivalent to the text
+        MessageComposerInput(
+            messageText = messageText,
+            onMessageTextChanged = { value ->
+                onMessageChanged(value)
+            },
+            messageComposerInputState = messageComposerState.messageComposeInputState,
+            onIsFocused = {
+                messageComposerState.toActive()
+            },
+            onNotFocused = {
+                messageComposerState.hasFocus = false
+            },
+            modifier = Modifier
+                .fillMaxWidth()
+                .then(
+                    when (messageComposerState.messageComposeInputState) {
+                        MessageComposeInputState.FullScreen ->
+                            Modifier
+                                .fillMaxHeight()
+                                .padding(end = dimensions().messageComposerPaddingEnd)
+                        MessageComposeInputState.Active -> {
+                            Modifier
+                                .heightIn(
+                                    max = dimensions().messageComposerActiveInputMaxHeight
+                                )
+                                .padding(
+                                    end = dimensions().messageComposerPaddingEnd
+                                )
+                        }
+                        else -> Modifier.wrapContentHeight()
+                    }
+                )
+        )
+    }
+}
+
+@ExperimentalAnimationApi
+@Composable
+private fun CollapseIconButtonBox(
+    transition: Transition<MessageComposeInputState>,
+    messageComposerState: MessageComposerInnerState
+) {
+    transition.AnimatedVisibility(visible = { state -> (state != MessageComposeInputState.Enabled) }) {
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
+        ) {
+            val collapseButtonRotationDegree by transition.animateFloat(
+                label = stringResource(R.string.animation_label_button_rotation_degree_transistion)
+            ) { state ->
+                when (state) {
+                    MessageComposeInputState.Active, MessageComposeInputState.Enabled -> 0f
+                    MessageComposeInputState.FullScreen -> 180f
+
                 }
             }
+            CollapseIconButton(
+                onCollapseClick = { messageComposerState.toggleFullScreen() },
+                collapseRotation = collapseButtonRotationDegree
+            )
         }
     }
 }
 
+@Composable
+private fun UserIsBlockedMessage() {
```

Comment:
As this is not really a conditional compose object, my suggestion would be to rename it to `BlockedUserMessage` and move it to a separate file.